src/shared/components/oncoprint/ResultsViewOncoprint.tsx

```diff
@@ -58,14 +58,16 @@ export interface IGenesetExpansionRecord {
 export const SAMPLE_MODE_URL_PARAM = "show_samples";
 export const CLINICAL_TRACKS_URL_PARAM = "clinicallist";
 export const HEATMAP_TRACKS_URL_PARAM = "heatmap_track_groups";
+export const TREATMENT_LIST_URL_PARAM = "treatment_list";
 
 const CLINICAL_TRACK_KEY_PREFIX = "CLINICALTRACK_";
 
 type HeatmapTrackGroupRecord = {
     trackGroupIndex:number,
-    genes:ObservableMap<boolean>,
+    molecularAlterationType:string,
+    entities:ObservableMap<boolean>,
```

Comment:
I really dont like storing genes and treatments in the same map - better to have one map for genes and one for treatments

Reply:
I personally disagree. The scope of `genetic_profiles` is expanding as cBioPortal evolves (genes->mRNA->protein, gene sets, treatments, ...). Heatmap tracks can be any of these so I think it is better to abandon the separation based on type. This also prevents the rather arbitrary priority that is imposed by cBioPortal relating to representation of heatmap tracks that are of different type w/o much complication. I assume that when selecting a treatment profile then a molecular profile that the user expects the treatment is show on top. When separating all profiles into different track group maps this is not the behavior that is designed for. When allowing other profile types besides molecular, I would argue to combine them.

Reply:
In this particular case, the issue is that gene identifiers and treatment identifiers don't live in the same universe. So I find it hacky to use them as keys in the same map.

Reply:
I have refactored the MobXPromisses for molecular profiles and treatment profiles. However, I am not quite understanding what your proposal is regarding your last comment:
>In this particular case, the issue is that gene identifiers and treatment identifiers don't live in the same universe. So I find it hacky to use them as keys in the same map.
The `HeatmapTrackGroupRecord` type is used by the molecularProfileIdToHeatmapTracks observable map of ResultsViewOncoprint. This map is keyed by molecular profile id:

`this.molecularProfileIdToHeatmapTracks.set(molecularProfileId, trackGroup);`

Although the molecular and treatment identifiers live in separate spaces, the molecular profile identifiers that are used for keying do live in the same space. I would be more that happy to accomodate your whish, but I am uncertain what your conceptual problem is and how I should solve it.

Reply:
Sorry, I should be more specific: it's the `entities` property that bothered me. I would prefer if you split that up into `genes` and `treatments`. However, I'm now understanding that for a single HeatmapTrackGroupRecord you would only have genes, or treatments, which I didn't understand before. In that case, it's fine this way, just please add a comment next to it indicating what specifically it's a map of - for genes, hugo gene symbols?  for treatments, treatment id?

Reply:
Comment added.